flake.nix

```diff
@@ -1,30 +1,58 @@
 {
   description = "Seabug";
+
   inputs = {
-    nixpkgs = { url = "github:NixOS/nixpkgs"; };
-    cardano-node.url = "github:input-output-hk/cardano-node/1.35.1";
-    flake-utils = { url = "github:numtide/flake-utils"; };
-    # https://github.com/hercules-ci/arion/pull/153
-    arion = {
-      url = "github:hercules-ci/arion/bd3e2fe4e372d0b5f965f25f27c5eb7c1a618c4a";
-    };
+    nixpkgs.url = github:NixOS/nixpkgs/nixos-unstable;
+
+    # TODO change after they are merged
+    nft-marketplace-frontend.url = github:mlabs-haskell/nft-marketplace/aciceri/contract-updates-and-nix;
+
+    nft-marketplace-server.url = github:mlabs-haskell/nft-marketplace-server/aciceri/nix;
   };
-  outputs = { self, nixpkgs, cardano-node, flake-utils, arion }:
-    flake-utils.lib.eachDefaultSystem (system:
-      let pkgs = nixpkgs.legacyPackages.${system};
-      in rec {
-        devShell = pkgs.mkShell {
-          nativeBuildInputs = with pkgs; [
-            arion.packages.${system}.arion
-            cardano-node.packages.${system}.cardano-cli
-            curl
-            httpie
-            ipfs
-            jq
-            postgresql
-            shfmt
-            expect
+  outputs =
+    { self
+    , nixpkgs
+    , nft-marketplace-frontend
+    , nft-marketplace-server
+    } @ inputs:
+    let
+      supportedSystems = [ "x86_64-linux" "x86_64-darwin" ];
+
+      pkgsFor = lib.genAttrs supportedSystems (system: nixpkgs.legacyPackages.${system});
+
+      lib = nixpkgs.lib.extend (self: super: {
+        perSystem = super.genAttrs supportedSystems;
+      });
+    in
+    {
+      nixosModules = {
+        seabug = {
+          imports = [
+            inputs.nft-marketplace-frontend.nixosModules.nft-marketplace-frontend
+            inputs.nft-marketplace-server.nixosModules.nft-marketplace-server
+            ./nix/seabug.nix
+          ];
+          nixpkgs.overlays = with inputs; [
+            nft-marketplace-frontend.overlays.nft-marketplace-frontend
+            nft-marketplace-server.overlays.nft-marketplace-server
           ];
         };
+        default = self.modules.seabug;
+      };
+
+      nixosConfigurations.seabug-vm = lib.nixosSystem {
```

Comment:
we should make it more clear that this is a test vm with root access and no password, let's name it `test-vm` and maybe add a comment here

Reply:
I'll do it